### DISCUSSION TITLE
What's the recommended way to fetch once a variable has loaded?

### DISCUSSION THREAD

**Discussion by 0xPT** [wevm/wagmi#1635] [Q&A]
I'm currently chaining a few reads/writes, and I'd prefer to not useEffect() everything and then .refetch() once I know I have the data for the wagmi hooks. Also, typescript gets real mad for the `args` param in the hooks because some of the args are potentially undefined when the component loads

Is the best way to go about this to utilize the onSuccess() function and set a state variable that keeps track of this stuff? Am I architecting my app in an anti-pattern to wagmi? Would love some cleaner suggestions.

Thank you!

**Comment by tmm** [comment]
Hard to tell without seeing code, but generally the best thing for dependent queries is to use the `enabled` config param. Setting it in an `onSuccess` callback seems fine.
